core/client/db/src/lib.rs

```diff
@@ -537,11 +585,10 @@ impl<Block> client::backend::BlockImportOperation<Block, Blake2Hasher>
 
 	fn update_storage(
 		&mut self,
-		update: StorageCollection,
-		child_update: ChildStorageCollection,
+		update: FullStorageCollection,
 	) -> ClientResult<()> {
-		self.storage_updates = update;
-		self.child_storage_updates = child_update;
+		self.storage_updates = update.top;
+		self.child_storage_updates = update.children;
```

Comment:
Yes that is part of the tasks I kept out of the pr (in the TODO list for storage_cache management (I always think of this method as the main point for state storage update, but it only really go to the cache update and the rpc)).
Change for kv where put alongside change for state in 'BlockImportOperation' db_update field. Maybe I should use a new field (using a pair helps ensuring I do not have some omission but at this point it could get split).
Thinking twice it could go through 'update_storage', it may be a better place to handle such changes.